frontend/test/metabase/scenarios/organization/timelines-question.cy.spec.js

```diff
@@ -257,6 +257,90 @@ describe("scenarios > organization > timelines > question", () => {
       cy.findByText("Visualization").should("be.visible");
       cy.findByLabelText("star icon").should("be.visible");
     });
+
+    it("should toggle individual event visibility", () => {
+      cy.createTimelineWithEvents({
+        timeline: { name: "Releases" },
+        events: [
+          { name: "RC1", timestamp: "2018-10-20T00:00:00Z", icon: "cloud" },
+        ],
+      });
+
+      cy.createTimelineWithEvents({
+        timeline: { name: "Timeline for collection", collection_id: 1 },
+        events: [
+          { name: "TC1", timestamp: "2016-05-20T00:00:00Z", icon: "warning" },
+        ],
+      });
+
+      visitQuestion(3);
+
+      cy.findByText("Visualization").should("be.visible");
+      cy.findByLabelText("cloud icon").should("be.visible");
+
+      // should hide individual events from chart if hidden in sidebar
+      cy.icon("calendar").click();
+      cy.findByText("Releases").click();
+      toggleEventVisibility("RC1");
+
+      cy.get(".x.axis").within(() => {
+        cy.findByLabelText("cloud icon").should("not.exist");
+      });
+
+      // should show individual events in chart again
+      toggleEventVisibility("RC1");
+
+      cy.get(".x.axis").within(() => {
+        cy.findByLabelText("cloud icon").should("be.visible");
+      });
+
+      // should show a newly created event
+      cy.button("Add an event").click();
+      cy.findByLabelText("Event name").type("RC2");
+      cy.findByLabelText("Date").type("10/20/2018");
+      cy.button("Create").click();
+      cy.wait("@createEvent");
+
+      cy.get(".x.axis").within(() => {
+        cy.findByLabelText("star icon").should("be.visible");
+      });
+
+      // should then hide the newly created event
+      toggleEventVisibility("RC2");
+
+      cy.get(".x.axis").within(() => {
+        cy.findByLabelText("star icon").should("not.exist");
+      });
+
+      // should initialize events in a hidden timelime
+      // with event checkboxes disabled and unchecked
+      cy.findByText("Timeline for collection").click();
+
+      cy.findByText("TC1")
+        .closest("[aria-label=Timeline event card]")
+        .within(() => {
+          cy.findByRole("checkbox").should("be.disabled");
+          cy.findByRole("checkbox").should("not.be.checked");
+        });
+
+      // making a hidden timeline visible
+      // should make its events automatically visible
+      cy.findByText("Timeline for collection")
+        .closest("[aria-label=Timeline card header]")
+        .within(() => cy.findByRole("checkbox").click());
+
+      cy.get(".x.axis").within(() => {
+        cy.findByLabelText("warning icon").should("be.visible");
```

Comment:
Nit: You can simply use `cy.icon("warning");`

Reply:
@nemanjaglumac 

> Nit: You can simply use cy.icon("warning").should("be.visible");

Thanks, updating.

Also, made this quick PR updating the calls in already-existing test cases in this file:
https://github.com/metabase/metabase/pull/27346

Reply:
@nemanjaglumac,

I've checked and we do need to use findByLabelText to select these icons in the chart.

These event icons are rendered as svg paths, and the class names we use to select with cy.icon are not added to them.